tests/utapi/awsNodeSdk.js

```diff
@@ -1,159 +1,194 @@
 }
+
 function deleteObjects(bucket, keys, cb) {
-    const objects = keys.map(key => {
-        const keyObj = {
-            Key: key,
-        };
-        return keyObj;
-    });
+    const objects = keys.map(key => ({ Key: key }));
+    const deleteRequest = { Objects: objects, Quiet: true };
     const params = {
         Bucket: bucket,
-        Delete: {
-            Objects: objects,
-            Quiet: true,
-        },
+        Delete: deleteRequest,
     };
-    return s3Client.deleteObjects(params, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    return s3Client.send(new DeleteObjectsCommand(params))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function copyObject(bucket, key, cb) {
-    return s3Client.copyObject({
-        Bucket: bucket,
-        CopySource: `/${bucket}/${key}`,
-        Key: `${key}-copy`,
-    }, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    const params = { Bucket: bucket, CopySource: `${bucket}/${key}`, Key: `${key}-copy` };
+    return s3Client.send(new CopyObjectCommand(params))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function enableVersioning(bucket, enable, cb) {
-    const versioningStatus = {
-        Status: enable ? 'Enabled' : 'Disabled',
-    };
-    return s3Client.putBucketVersioning({
-        Bucket: bucket,
-        VersioningConfiguration: versioningStatus,
-    }, err => {
-        assert.ifError(err);
-        return cb(err);
-    });
+    const versioningStatus = { Status: enable ? 'Enabled' : 'Disabled' };
+    const params = { Bucket: bucket, VersioningConfiguration: versioningStatus };
+    return s3Client.send(new PutBucketVersioningCommand(params))
+        .then(() => cb())
+        .catch(cb);
 }
+
 function deleteVersionList(versionList, bucket, callback) {
     if (versionList === undefined || versionList.length === 0) {
         return callback();
     }
-    const params = { Bucket: bucket, Delete: { Objects: [] } };
+    const deleteRequest = { Objects: [] };
     versionList.forEach(version => {
-        params.Delete.Objects.push({
-            Key: version.Key, VersionId: version.VersionId,
-        });
+        deleteRequest.Objects.push({ Key: version.Key, VersionId: version.VersionId });
     });
-
-    return s3Client.deleteObjects(params, callback);
+    const params = {
+        Bucket: bucket,
+        Delete: deleteRequest,
+    };
+    return s3Client.send(new DeleteObjectsCommand(params))
+        .then(() => callback())
+        .catch(callback);
 }
+
 function removeAllVersions(params, callback) {
     const bucket = params.Bucket;
     async.waterfall([
-        cb => s3Client.listObjectVersions(params, cb),
-        (data, cb) => deleteVersionList(data.DeleteMarkers, bucket,
-            err => cb(err, data)),
-        (data, cb) => deleteVersionList(data.Versions, bucket,
-            err => cb(err, data)),
+        cb => s3Client.send(new ListObjectVersionsCommand(params))
```

Comment:
```suggestion
        async() => s3Client.send(new ListObjectVersionsCommand(params)),
```

Async can understand the async/await pattern